keystore/src/connection/platform/wasm/storage.rs

```diff
@@ -131,29 +129,90 @@ impl WasmEncryptedStorage {
         }
     }
 
+    /// Copied from Rexie.
+    pub(crate) async fn scan(
+        object_store: &ObjectStore,
+        key_range: Option<KeyRange>,
+        limit: Option<u32>,
+        offset: Option<u32>,
+        direction: Option<CursorDirection>,
+    ) -> CryptoKeystoreResult<Vec<(JsValue, JsValue)>> {
+        let cursor = object_store.open_cursor(key_range.map(Into::into), direction)?.await?;
+
+        match cursor {
+            None => Ok(Vec::new()),
+            Some(cursor) => {
+                let mut cursor = cursor.into_managed();
+
+                let mut result = Vec::new();
+
+                match limit {
+                    Some(limit) => {
+                        if let Some(offset) = offset {
+                            cursor.advance(offset).await?;
+                        }
+
+                        for _ in 0..limit {
+                            let key = cursor.key()?;
+                            let value = cursor.value()?;
+
+                            match (key, value) {
+                                (Some(key), Some(value)) => {
+                                    result.push((key, value));
+                                    cursor.next(None).await?;
+                                }
+                                _ => break,
+                            }
+                        }
+                    }
+                    None => {
+                        if let Some(offset) = offset {
+                            cursor.advance(offset).await?;
+                        }
+
+                        loop {
+                            let key = cursor.key()?;
+                            let value = cursor.value()?;
+
+                            match (key, value) {
+                                (Some(key), Some(value)) => {
+                                    result.push((key, value));
+                                    cursor.next(None).await?;
+                                }
+                                _ => break,
+                            }
+                        }
+                    }
+                }
```

Comment:
I know this code is copied from Rexie, but I can't shake the feeling that we could do better.

For instance, something like the below:
```rust
    let it = std::iter::from_fn(|| {
         let key = cursor.key()?;
         let value = cursor.value()?;
         
         if key.is_some() {
            cursor.next(None).await?
            Some((key, value))
         } else {
            None
         }
    });
    if let Some(limit) = limit {
        it.take(limit).collect::<Vec<_>>()
    } else {
        it.collect::<Vec<_>>()
    }
```

Reply:
Looks like this would only work with async iterators, right? Unfortunately they're still unstable